features/lwipstack/LWIPInterfaceL3IP.cpp

```diff
@@ -0,0 +1,167 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "lwip/tcpip.h"
+#include "lwip/tcp.h"
+#include "lwip/ip.h"
+#include "netif/etharp.h"
+#include "lwip/ethip6.h"
+#include "netsocket/nsapi_types.h"
+#include "netsocket/L3IP.h"
+
+#include "LWIPStack.h"
+
+#if LWIP_L3IP
+
+err_t LWIP::Interface::l3ip_low_level_output(struct netif *netif, struct pbuf *p, const ip4_addr_t *ipaddr)
+{
+    /* Increase reference counter since lwip stores handle to pbuf and frees
+       it after output */
+    pbuf_ref(p);
+
+    LWIP::Interface *mbed_if = static_cast<LWIP::Interface *>(netif->state);
+    bool ret = mbed_if->l3ip->link_out(p);
+    return ret ? ERR_OK : ERR_IF;
+}
+
+void LWIP::Interface::l3ip_input(net_stack_mem_buf_t *buf)
+{
+    struct pbuf *p = static_cast<struct pbuf *>(buf);
+
+    /* pass all packets to ethernet_input, which decides what packets it supports */
```

Comment:
Not to ethernet input here? Should this be IP stack input.

Reply:
fixed